### ISSUE TITLE
Consolidate the file upload widget value metadata and content?

### ISSUE THREAD

**Issue by jasongrout** [jupyter-widgets/ipywidgets#2721]
We've recently discussed the format for the file upload value attribute. It looks like what was settled on was something like: 
```python
[{'metadata': {'name': 'Untitled1.ipynb',
   'type': '',
   'size': 1734,
   'lastModified': 1552316652863},
  'content': <memory at 0x10eeadd00>}]
```
In playing around with it a bit, it seems much simpler to just combine it into one dictionary, sort of like how the [Jupyter contents api](http://petstore.swagger.io/?url=https://raw.githubusercontent.com/jupyter/notebook/master/notebook/services/api/api.yaml#/contents/get_api_contents__path_) does it:
```python
[{'name': 'Untitled1.ipynb',
   'type': '',
   'size': 1734,
   'lastModified': 1552316652863,
   'content': <memory at 0x10eeadd00>}]
```
I think it's simpler to not have to go through the metadata key, like:

```python
for f in widget.value:
    print(f['name'])
    print(f['size'])
    print(f["content"].tobytes().decode("utf-8"))
```
Thoughts?

**Comment by pbugnion** [comment]
I agree having a single top-level dictionary is simpler. I think this was just forgotten about when we pushed #2666.

**Comment by jasongrout** [comment]
Also, I notice in the documentation from #2714, we have an example of converting the contents to a string: `uploaded_file["content"].tobytes().decode("utf-8")`. As far as I can tell, this creates an extra copy (`tobytes` does a copy). It would probably be better to do:

```python
import codecs
codec.decode(uploaded_file["content"], 'utf-8')
```
which does not do an extra copy by working on the memory view directly.

**Comment by pbugnion** [comment]
I've addressed the docs issue in #2722.

**Comment by jtpio** [comment]
Agree a single dict is simpler.

PR incoming.

**Comment by pbugnion** [comment]
Fixed by #2724 and #2722.
